### ISSUE TITLE
Bug: RetrieveBlocks does not return error when no blocks found

### ISSUE THREAD

**Issue by mastergaurang94** [rollkit/rollkit#1540]
### Version of Git SHA

gm (Main Branch): SHA d658b1fac2c67853245f2aac6df5ef63c41289f6 (init_local.sh)
rollkit (Main Branch): SHA e014d85015c427a33b47ea2f17e60d8a38c9cc67
go-da: v0.2.0
celestia: local-celestia-devnet:v0.12.7

### System OS

MacOS

### Others

Adjusted [init-local.sh ](https://github.com/rollkit/docs/blob/main/scripts/gm/init-local.sh#L90) to print log_level debug.

### Steps to reproduce it

1. Start celestia node via `docker run -t -i \
    -p 26650:26650 -p 26657:26657 -p 26658:26658 -p 26659:26659 -p 9090:9090 \
    ghcr.io/rollkit/local-celestia-devnet:v0.12.7`
2. Followed tutorial (no deviations) and ran [init_local.sh](https://raw.githubusercontent.com/rollkit/docs/main/scripts/gm/init-local.sh).

Noticed a few things:
a. Celestia node fails to find blob
```ERROR	rpc	go-jsonrpc@v0.3.1/handler.go:484	error and res returned	{"request": {"jsonrpc":"2.0","id":37872,"method":"blob.GetAll","params":[32564,["AAAAAAAAAAAAAAAAAAAAAAAAAAAA1HlbHGhasIQ="]],"meta":{"SpanContext":"AACVKw+4AkKpqNaFOC7CP3K0ARkZul9TTZjIAgA="}}, "r.err": "getting blobs for namespace(000000000000000000000000000000000000000000d4795b1c685ab084): blob: not found\nblob: not found", "res": []}```
b. There are no corresponding logs in gmd rollup about blobNotFound.
c. Also, sequencer does not produce blocks. When attempting a tx via `gmd tx bank send` it fails with "rpc error: code = Unknown desc = gm is not ready; please wait for first block: invalid height".
 

### Expected result

A few things:

1) Rollkit should print out this debug warning log, but does not enter: https://github.com/rollkit/rollkit/blob/main/block/manager.go#L570.

2) Sequencer should produce blocks, including first block.

### Actual result

For the above issues, respectively:

1) Gm rollup continues and prints the following:
```
8:03PM DBG retrieved potential blocks daHeight=34919 module=BlockManager n=0
8:03PM DBG trying to retrieve block from DA daHeight=34920 module=BlockManager
```

2) No blocks produced. For instance, when running restart_local.sh which just `gmd start`, the following error is given: `error while initializing BlockManager: genesis.InitialHeight (1) is greater than last stored state's LastBlockHeight (0)`. No other logs indicating block production.

### Notes

Regarding the above issues, I have looked into the first and have a proposal for the fix. 

### First issue - no log in rollup for no blocks
TL;DR: In da.go, this [check](https://github.com/rollkit/rollkit/blob/main/da/da.go#L166) needs to be `if len(ids) == 0` to properly handle an initialized empty slice. Will make a PR w/ test if there are no objections since it's a simple fix, but wanted to document here.

I confirmed it works by adjusting in my local environment. The gRPC call comes from celestia-da, which I confirmed accurately returns a `nil, nil`: https://github.com/rollkit/celestia-da/blob/5892064d5539177b97d5ac406eb0def6e0fc4bf5/celestia/celestia.go#L76. Notice also the gRPC call above has `res: []` instead of nil as response.

It then is processed in go-da/proxy/client.go: https://github.com/rollkit/go-da/blob/main/proxy/client.go#L70. As `resp, err` are both nil, it proceeds into `idsPB2DA(resp.Ids)` with nil IDs. A slice is initialized and returned: https://github.com/rollkit/go-da/blob/main/proxy/util.go#L25.

That results in an initialized empty slice in da.go (value `[]` with type `[][]uint8`, which requires `len(ids) == 0` to be caught. I tested out the fix in my local and here is the resulting log showing up now: 
```
8:19PM DBG no block found daHeight=35384 module=BlockManager reason="blob: not found"
8:19PM DBG trying to retrieve block from DA daHeight=35385 module=BlockManager
```

### Second Issue - sequencer not producing blocks

I can separate this out into another issue once I've investigated it a bit more.

**Comment by mastergaurang94** [comment]
I have a local branch with the change + tests for BlobNotFound. Would I be able to get access to push up branches (at least)?

**Comment by gupadhyaya** [comment]
@mastergaurang94 we had a bug in init-local.sh, which should be fixed now: https://github.com/rollkit/docs/pull/322/

**Comment by gupadhyaya** [comment]
agree on modifying the check `ids == nil` to `len(ids) == 0`. please make a PR.

**Comment by mastergaurang94** [comment]
@gupadhyaya ah, makes sense. Have branch ready, but get 403 when pushing up. Need access in rollkit repo.

**Comment by gupadhyaya** [comment]
> @gupadhyaya ah, makes sense. Have branch ready, but get 403 when pushing up. Need access in rollkit repo.

fork rollkit, push changes to a feature branch of your fork and then make a PR from your fork to rollkit repo. that should work.
